gdal/frmts/raw/byndataset.cpp

```diff
@@ -0,0 +1,766 @@
+    poDS->nRasterXSize = (GInt32) ( ( ( dfEast  - dfWest  ) / dfDLon ) + 1.0 );
+
+    if (!GDALCheckDatasetDimensions(poDS->nRasterXSize, poDS->nRasterYSize))
+    {
+        delete poDS;
+        return nullptr;
+    }
+
+    /*****************************/
+    /* Build GeoTransform matrix */
+    /*****************************/
+
+    poDS->adfGeoTransform[0] = ( dfWest - ( dfDLon / 2.0 ) ) / 3600.0;
+    poDS->adfGeoTransform[1] = dfDLon / 3600.0;
+    poDS->adfGeoTransform[2] = 0.0;
+    poDS->adfGeoTransform[3] = ( dfNorth + ( dfDLat / 2.0 ) ) / 3600.0;
+    poDS->adfGeoTransform[4] = 0.0;
+    poDS->adfGeoTransform[5] = -1 * dfDLat / 3600.0;
+
+    CPLDebug("BYN","South           = %d",poDS->hHeader.nSouth);
+    CPLDebug("BYN","North           = %d",poDS->hHeader.nNorth);
+    CPLDebug("BYN","West            = %d",poDS->hHeader.nWest);
+    CPLDebug("BYN","East            = %d",poDS->hHeader.nEast);
+    CPLDebug("BYN","DLat            = %d",poDS->hHeader.nDLat);
+    CPLDebug("BYN","DLon            = %d",poDS->hHeader.nDLon);
+    CPLDebug("BYN","DGlobal         = %d",poDS->hHeader.nGlobal);
+    CPLDebug("BYN","DType           = %d",poDS->hHeader.nType);
+    CPLDebug("BYN","Factor          = %f",poDS->hHeader.dfFactor);
+    CPLDebug("BYN","SizeOf          = %d",poDS->hHeader.nSizeOf);
+    CPLDebug("BYN","VDatum          = %d",poDS->hHeader.nVDatum);
+    CPLDebug("BYN","Decription      = %d",poDS->hHeader.nDescrip);
+    CPLDebug("BYN","SubType         = %d",poDS->hHeader.nSubType);
+    CPLDebug("BYN","Datum           = %d",poDS->hHeader.nDatum);
+    CPLDebug("BYN","Ellipsoid       = %d",poDS->hHeader.nEllipsoid);
+    CPLDebug("BYN","ByteOrder       = %d",poDS->hHeader.nByteOrder);
+    CPLDebug("BYN","Scale           = %d",poDS->hHeader.nScale);
+    CPLDebug("BYN","Wo              = %f",poDS->hHeader.dfWo);
+    CPLDebug("BYN","GM              = %f",poDS->hHeader.dfGM);
+    CPLDebug("BYN","TideSystem      = %d",poDS->hHeader.nTideSys);
+    CPLDebug("BYN","RefRealzation   = %d",poDS->hHeader.nRealiz);
+    CPLDebug("BYN","Epoch           = %f",poDS->hHeader.dEpoch);
+    CPLDebug("BYN","PtType          = %d",poDS->hHeader.nPtType);
+    CPLDebug("BYN","RasterXSize     = %d ", poDS->nRasterXSize );
+    CPLDebug("BYN","RasterYSize     = %d ", poDS->nRasterYSize );
+    CPLDebug("BYN","GeoTransform[0] = %f ", poDS->adfGeoTransform[0] );
+    CPLDebug("BYN","GeoTransform[1] = %f ", poDS->adfGeoTransform[1] );
+    CPLDebug("BYN","GeoTransform[2] = %f ", poDS->adfGeoTransform[2] );
+    CPLDebug("BYN","GeoTransform[3] = %f ", poDS->adfGeoTransform[3] );
+    CPLDebug("BYN","GeoTransform[4] = %f ", poDS->adfGeoTransform[4] );
+    CPLDebug("BYN","GeoTransform[5] = %f ", poDS->adfGeoTransform[5] );
+
+    /*****************/
+    /* Data type     */
+    /*****************/
+ 
+    GDALDataType eDT = GDT_Unknown;
+
+    if ( poDS->hHeader.nSizeOf == 2 )
+       eDT = GDT_Int16;
+    else
```

Comment:
missing { }

Reply:
I didn't get that one. Is that what you mean:

    if ( poDS->hHeader.nSizeOf == 2 )
       eDT = GDT_Int16;
    else
    { <==== missing brakets
        if ( poDS->hHeader.nSizeOf == 4 )
            eDT = GDT_Int32;
        else
        {   
            delete poDS;
            return nullptr;
        }
    } <==== missing brakets

Maybe you would preferer a fake-C-elseif :

    if ( poDS->hHeader.nSizeOf == 2 )
       eDT = GDT_Int16;
    else if ( poDS->hHeader.nSizeOf == 4 )
            eDT = GDT_Int32;
    else
    {   
        delete poDS;
        return nullptr;
    }

It looks like it a matter of style but I appreciate your suggestion to revise it.

Reply:
Your option 2 with consistent indentation would be better in fact. And always putting {} even for single-line clause tends to be better to avoid bugs.

```
if ( poDS->hHeader.nSizeOf == 2 )
{
    eDT = GDT_Int16;
}
else if ( poDS->hHeader.nSizeOf == 4 )
{
    eDT = GDT_Int32;
}
else
{   
    delete poDS;
    return nullptr;
}
```

Reply:
I used to put brackets and begin-end for single-line blocks too, but I recently came across an author who made me change my mind.
I am almost done with all the changes. Thanks for the review.